src/actions.js

```diff
@@ -127,11 +127,15 @@ export const cut = (state: Types.StoreState<*>) => ({
   cut: true
 });
 
-export function paste<Cell: Types.CellBase>(state: Types.StoreState<Cell>) {
-  if (PointSet.isEmpty(state.copied)) {
-    return null;
-  }
-  const minPoint = PointSet.min(state.copied);
+export async function paste<Cell: Types.CellBase>(
```

Comment:
Actions should not do IO. Please return the readTextFromClipboard to the component.